kong/clustering/services/sync/rpc.lua

```diff
@@ -363,45 +368,69 @@ local function sync_handler(premature)
     return
   end
 
-  local res, err = concurrency.with_worker_mutex(SYNC_MUTEX_OPTS, function()
-    -- `do_sync()` is run twice in a row to report back new version number
-    -- to CP quickly after sync. (`kong.sync.v2.get_delta` is used for both pulling delta
-    -- as well as status reporting)
-    for _ = 1, 2 do
-      local ok, err = do_sync()
-      if not ok then
-        return nil, err
-      end
-    end -- for
-
-    return true
-  end)
+  local res, err = concurrency.with_worker_mutex(SYNC_MUTEX_OPTS, do_sync)
   if not res and err ~= "timeout" then
     ngx_log(ngx_ERR, "unable to create worker mutex and sync: ", err)
   end
 end
 
 
-local function start_sync_timer(timer_func, delay)
-  local hdl, err = timer_func(delay, sync_handler)
+local sync_once
 
-  if not hdl then
+
+local function start_sync_once_timer(retry_count)
+  local ok, err = ngx.timer.at(0, sync_once, retry_count or 0)
+  if not ok then
     return nil, err
   end
 
   return true
 end
 
 
+function sync_once(premature, retry_count)
+  if premature then
+    return
+  end
+
+  sync_handler()  
```

Comment:
should remove `premature` in sync_handler()

Reply:
No. It's used in `ngx.timer.every`